os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -104,24 +105,37 @@ typedef struct {
 } inout_struc;
 
 typedef struct {
-	uint32_t factorykey_device_idx;
-	uint32_t factorykey_cert_idx;
 	uint32_t factory_cert_addr;
 	uint32_t factory_key_addr;
 	uint32_t sstorage_addr;
 	uint32_t samsung_key_addr;
+	hal_key_type factory_slot_key_type[USABLE_FACTORY_KEY_INDEX];
 } factory_struc;
 
 static uint8_t *ns_buf = NULL;
-
 extern void *rtl_set_ns_func(void);
 
+/**
+ * Helper function
+ */
+static int se_ameba_index_chk(uint32_t key_idx)
+{
+	int ret = HAL_SUCCESS;
+
+	/* Check if in valid Factory range */
+	if ((key_idx >= USABLE_FACTORY_KEY_INDEX) && (key_idx < FACTORY_KEY_INDEX_MAX)) {
+		ret = HAL_INVALID_SLOT_RANGE;
+	}
+
+	return ret;
+}
 
 int se_ameba_hal_init(hal_init_param *params)
 {
 	AWRAP_ENTER;
 
 	int ret;
+	int count;
```

Comment:
how about to initialize the variable. because code analysis tool warn uninitialized variable

Reply:
Done